### PR TITLE
Dex trades legacy seed

### PR DESCRIPTION
**Pull request by 0xRobin**
This adds a legacy version of the dex trades seed test (and seed itself)

### CODE REVIEW

**Review by jkylling — APPROVED** [review]

**Review by jeff-dude — APPROVED** [review]
@couralex6 was messing with this a bit too during uniswap migration. just tagging to get his approval and awareness.

**Review by jeff-dude — APPROVED** [review]
this assumes @couralex6 has the new `tests/generic/check_dex_seed.sql` added when migrating (and translated)? so tests pass in CI?

then since we aren't modifying schema files, we won't ever use `check_dex_seed_legacy` test?

**Comment by couralex6** [comment]
I already have most of these changes in #3667 . Looks like I'm missing the some `'dex_trades_seed'` -> `'dex_trades_seed_legacy'`. Do you mind if I just import those in the uniswap PR?

**Comment by 0xRobin** [comment]
> then since we aren't modifying schema files, we won't ever use `check_dex_seed_legacy` test?

correct for `dunesql` models, `check_dex_seed_legacy` will still be used for non-migrated spark models.

**Comment by 0xRobin** [comment]
> Do you mind if I just import those in the uniswap PR?

you can definitely merge this into your PR, but I still suggest we merge this one into main asap as there are other PRs that need this change.

File: tests/generic/check_dex_seed_legacy.sql
Comment (jeff-dude) [inline tests/generic/check_dex_seed_legacy.sql:1]: this is what i was referring to -- we don't have the dunesql version of this macro in this PR? is that coming in another PR?

Comment (jeff-dude) [reply tests/generic/check_dex_seed_legacy.sql:1]: we have the seed itself, but not the test

Comment (0xRobin) [reply tests/generic/check_dex_seed_legacy.sql:1]: oh @jeff-dude I get your comment now!

we already have both `check_dex_seed` (dunesql) and `check_dex_seed_legacy` on main.
https://github.com/duneanalytics/spellbook/blob/main/tests/generic/check_dex_seed.sql

They where just using both the same seed which was an initial oversight.

Comment (jeff-dude) [reply tests/generic/check_dex_seed_legacy.sql:1]: ahhh okay, my fault, thank you

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
merging now @couralex6

**Pull request merged**